src/power_grid_model/utils.py

```diff
@@ -54,6 +55,24 @@ def _get_component_scenario(component_scenarios: BatchArray) -> np.ndarray:
     return {component: _get_component_scenario(component_data) for component, component_data in dataset.items()}
 
 
+def get_dataset_batch_size(dataset: BatchDataset) -> int:
```

Comment:
a `BatchDataset` type is not a `CConstDataset`. in fact, as I mentioned before, the `CConstDataset` (or in general, any class prefixed with `C`) is low-level and should not be exposed to the user.

If there's a really good reason to keep it, please update the type as such:

```suggestion
def get_dataset_batch_size(dataset: Union[BatchDataset, CConstDataset]) -> int:
```